scripts/gulpfiles/release_tasks.js

```diff
@@ -0,0 +1,182 @@
+  if (!majorVersion) {
+    done(new Error('Something went wrong when getting the major version number.'));
+  } else if (!updateType) {
+    // User selected to cancel.
+    done(new Error('Cancelling process.'));
+  }
+
+  switch (updateType.toLowerCase()) {
+    case 'major':
+      majorVersion++;
+      execSync(`npm --no-git-tag-version version ${majorVersion}.$(date +'%Y%m%d').0`, {stdio: 'inherit'});
+      done();
+      break;
+    case 'minor':
+      execSync(`npm --no-git-tag-version version ${majorVersion}.$(date +'%Y%m%d').0`, {stdio: 'inherit'});
+      done();
+      break;
+    case 'patch':
+      execSync(`npm --no-git-tag-version version patch`, {stdio: 'inherit'});
+      done();
+      break;
+    default:
+      done(new Error('In unexpected place in switch statement.'))
+  }
+}
+
+// Prompt the user to figure out what kind of version update we should do.
+function updateVersionPrompt(done) {
+  var releaseTypes = ['Major', 'Minor', 'Patch'];
+  var index = readlineSync.keyInSelect(releaseTypes, 'Which version type?');
+  updateVersion(done, releaseTypes[index]);
+}
+
+// Checks with the user that they are on the correct git branch.
+function checkBranch(done) {
+  var gitBranchName = execSync('git rev-parse --abbrev-ref HEAD').toString();
+  if (readlineSync.keyInYN(`You are on '${gitBranchName.trim()}'. Is this the correct branch?`)) {
+    done();
+  } else {
+    done(new Error('Not on correct branch'));
+  }
+}
+
+
+// Sanity check that the dist folder exists, and that certain files are in the dist folder.
+function checkDist(done) {
+  const sanityFiles = ['blockly_compressed.js', 'blocks_compressed.js', 'core', 'blocks', 'generators'];
+  // Check that dist exists.
+  if (fs.existsSync(RELEASE_DIR)) {
+    // Sanity check that certain files exist in dist.
+    sanityFiles.forEach((fileName) => {
+      if (!fs.existsSync(`${RELEASE_DIR}/${fileName}`)) {
+        done(new Error(`Your dist folder does not contain:${fileName}`));
+      }
+    });
+    done();
+  } else {
+    done(new Error('The dist directory does not exist. Is npm run package being run?'));
+  }
+}
+
+// Check with the user that the version number is correct, then login and publish to npm.
+function loginAndPublish_(done, isBeta) {
+  delete require.cache[require.resolve('../../dist/package.json')]
+  var { version } = require('../../dist/package.json');
+  if(readlineSync.keyInYN(`You are about to publish blockly with the version number:${version}. Do you want to continue?`)) {
+    execSync(`npm login --registry https://wombat-dressing-room.appspot.com`, {stdio: 'inherit'});
+    execSync('npm pack', {cwd: RELEASE_DIR, stdio: 'inherit'});
+    console.log(`npm publish --registry https://wombat-dressing-room.appspot.com ${isBeta ? '--tag beta' : ''}`);
+    // TODO: Uncomment this line and test.
+    // execSync('npm publish --registry https://wombat-dressing-room.appspot.com', {cwd: RELEASE_DIR, stdio: 'inherit'});
```

Comment:
Can this TODO be removed or is this something that still needs to be tested?

Reply:
This still needs to be here. I am waiting for when we need to publish a patch, or beta version so that I can test.

Reply:
ok. Could you change to "Uncomment this line and test the next time we publish a patch, or beta version." and/or maybe create an issue that can be linked.
Then others with less context know what that's in there for.

Reply:
Just tested, so uncommenting this line and removed the npm pack above.